packages/circuits/helpers/timestamp.circom

```diff
@@ -0,0 +1,102 @@
+pragma circom 2.1.6;
+
+include "circomlib/circuits/comparators.circom";
+
+function numStringToNum(num) {
+    return num - 48;
+}
+
+// Converts a date string of format YYYYMMDDHHMMSS to a unix time
+// Assumes the input time is in UTC
+// includeHours - 1 to include hours, 0 to round down to day
+// includeMinutes - 1 to include minutes, 0 to round down to hour
+// includeSeconds - 1 to include seconds, 0 to round down to minute
+template DateStringToTimestamp(maxYears, includeHours, includeMinutes, includeSeconds) {
+    signal input in[14];
+    signal output out;
+
+    signal daysTillPreviousMonth[12] <== [0, 31, 59, 90, 120, 151, 181, 212, 243, 273, 304, 334];
+
+    var year = numStringToNum(in[0]) * 1000 + numStringToNum(in[1]) * 100 + numStringToNum(in[2]) * 10 + numStringToNum(in[3]);
+    var month = numStringToNum(in[4]) * 10 + numStringToNum(in[5]);
+    var day = numStringToNum(in[6]) * 10 + numStringToNum(in[7]);
```

Comment:
I think we should change to: 
signal year <== numStringToNum(in[0]) * 1000 + numStringToNum(in[1]) * 100 + numStringToNum(in[2]) * 10 + numStringToNum(in[3])

numStringToNum must be component also. 
Because with current implement `year` is independence with signal `in[0], in[1], in[2], in[3]` . 

Same with month and day.

Reply:
Maybe I wrong! I need check it again

Reply:
I did think about it - but from what I understood, since the variable is assigned to a signal later in the circuit  essentially making the input (transformed) bind to the signal, this is ok? 
We dont necessarily want year to be dependent on the signal but the values in daysPassed should be.
I could be wrong - I will change it if you think this is under-constrained. I checked with `circom inspect` earlier and didn't see any warnings on these.

Reply:
I did think about it - but from what I understood, since the variable is assigned to a signal later in the circuit essentially making the input (transformed) bind to the signal, this is ok?
> I recheck it! I think you are correct. But I still worry about this.

Reply:
I think I will change it, just to be safe. Number of constrains are same either way.